src/timeline-change-controller.js

```diff
@@ -0,0 +1,46 @@
+import videojs from 'video.js';
+
+/**
+ * The TimelineChangeController acts as a source for segment loaders to listen for and
+ * keep track of latest and pending timeline changes. This is useful to ensure proper
+ * sync, as each loader may need to make a consideration for what timeline the other
+ * loader is on before making changes which could impact the other loader's media.
+ *
+ * @class TimelineChangeController
+ * @extends videojs.EventTarget
+ */
+export default class TimelineChangeController extends videojs.EventTarget {
+  constructor() {
+    super();
+
+    this.pendingTimelineChanges_ = {};
+    this.lastTimelineChanges_ = {};
+  }
+
+  clearPendingTimelineChange(type) {
+    delete this.pendingTimelineChanges_[type];
```

Comment:
Definitely can. Any reason for the preference, or is it just more standard?

Reply:
Historically, delete causes performance issues since it changes the internal data structure where-as setting to null doesn't. I do wonder how much of an issue it still is. Might better more the more used a thing is.